### DISCUSSION TITLE
Is there a way to opt for json.RawMessage over map[string]interface{}?

### DISCUSSION THREAD

**Discussion by Ayo-Awe** [oapi-codegen/oapi-codegen#1778] [Q&A] [answered]
Is there a way to opt out of the default map[string]interface{} for dynamic objects in oapi-codegen. I have a use case where I'd like to have json.RawMessage over the map[string]interface{}

At the moment, I have to marshal the response back to bytes before storing it and unmarshal every time I pull it from the db. 

<img width="895" alt="image" src="https://github.com/user-attachments/assets/7348acd6-46bd-4f5a-9dc2-6f41bc086d55">

**Comment by jamietanna** [comment, marked as answer]
Interesting!

Using `x-go-type` via https://github.com/oapi-codegen/oapi-codegen#openapi-extensions should give you the ability to override it on a per-field level if that helps?

**Comment by Ayo-Awe** [comment]
It worked!!!
Thanks @jamietanna
